sdks/python/apache_beam/test_pipeline.py

```diff
@@ -0,0 +1,75 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test Pipeline, a wrapper of Pipeline for test purpose"""
+
+import argparse
+
+from apache_beam.pipeline import Pipeline
+from apache_beam.utils.options import PipelineOptions
+
+
+class TestPipeline(Pipeline):
+  """TestPipeline class can be used inside of tests that can be configured to
+  run against pipeline runner.
+
+  It has a functionality to parse arguments from command line and build pipeline
+  options for tests who runs against a pipeline runner and utilizes resources
+  of the pipeline runner. Those test functions are recommended to be tagged by
+  @pytest.mark.ValidatesRunner annotation.
+
+  Test functions that run against a pipeline runner are recommended to be tagged
+  by @pytest.mark.ValidatesRunner annotation. TestPipeline class has a
+  functionality to parse arguments from command line and build pipeline options
+  for such tests.
+
+  In order to configure the test with customized pipeline options from command
+  line, system argument 'test_options' can be used to obtains a list of pipeline
```

Comment:
I do not fully understand how 'test_options' work or what is a system argument. Maybe you can add an example of that here.